src/Domain/Structure/Direction.cpp

```diff
@@ -7,56 +7,81 @@
 #include <pup.h>
 #include <pup_stl.h>
 
+#include "Domain/Structure/Side.hpp"
 #include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
 #include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
 
 template <size_t VolumeDim>
-void Direction<VolumeDim>::pup(PUP::er& p) {
-  size_t version = 0;
-  p | version;
-  // Remember to increment the version number when making changes to this
-  // function. Retain support for unpacking data written by previous versions
-  // whenever possible. See `Domain` docs for details.
-  if (version >= 0) {
-    p | axis_;
-    p | side_;
-  }
+Direction<VolumeDim> Direction<VolumeDim>::self() {
+  return Direction<VolumeDim>{Axis::Xi, Side::Self};
 }
 
-template <>
-Direction<1>::Direction(const size_t dimension, const Side side) {
-  ASSERT(
-      0 == dimension,
-      "dim = " << dimension << ", for Direction<1> only dim = 0 is allowed.");
-  axis_ = Axis::Xi;
-  side_ = side;
+template <size_t VolumeDim>
+uint8_t Direction<VolumeDim>::self_bits() {
+  return static_cast<uint8_t>(Axis::Xi) bitor static_cast<uint8_t>(Side::Self);
```

Comment:
[optional] `return self().bits();` to reduce duplication.  (Or even get rid of this function entirely.)